hspec-core/src/Test/Hspec/Core/Runner.hs

```diff
@@ -82,10 +91,18 @@ applyDryRun c
       NodeWithCleanup _ xs -> NodeWithCleanup (\() -> return ()) (map removeCleanup xs)
       leaf@(Leaf _) -> leaf
 
--- | Run given spec and write a report to `stdout`.
+-- | Run a given spec and write a report to `stdout`.
 -- Exit with `exitFailure` if at least one spec item fails.
+--
+-- /Note/: `hspec` handles command-line options and reads config files.  This
+-- is not always desired.  Use `runSpec` if you need more control over these
+-- aspects.
 hspec :: Spec -> IO ()
-hspec = hspecWith defaultConfig
+hspec spec =
```

Comment:
What's wrong with the old implementation?  How was it different?

Reply:
I think we can deprecate `hspecWith` and friends down the road and only have `hspec` and `runSpec`.  For that reason, if somebody looks at the implementation of these I want them to see a future proof way to achieve the same.